h/admin/views/users.py

```diff
@@ -69,23 +63,56 @@ def users_activate(request):
                                     _query=(('username', user.username),)))
 
 
+@view_config(route_name='admin_users_rename',
+             request_method='POST',
+             permission='admin_users')
+def users_rename(request):
+    user = _form_request_user(request)
+
+    if user is None:
+        return httpexceptions.HTTPFound(
+            location=request.route_path('admin_users'))
+
+    old_username = user.username
+    new_username = request.params.get('new_username')
+
+    try:
+        svc = request.find_service(name='rename_user')
+        svc.check(old_username, new_username)
+
+        worker.rename_user.delay(old_username, new_username)
+
+        request.session.flash(
+            'The user "%s" will be renamed to "%s" in the backgroud. Refresh this page to see if it\'s already done' %
+            (old_username, new_username), 'success')
+
+        return httpexceptions.HTTPFound(
+            location=request.route_path('admin_users',
+                                        _query=(('username', new_username),)))
```

Comment:
This will do for now, but if there is a way we can be smarter about this and provide a better experience that would be great.

As an example I could just throw out there - most rename operations will take less than the 5 second request threshold. Could we wait for the operation to complete up to some time threshold and tell the user it succeeded if it finishes within that time, or show a "your username is being changed" message otherwise?